### PR TITLE
BOM-1581

### PR DESCRIPTION
**Pull request by awais786**
https://openedx.atlassian.net/browse/BOM-1581

Remove django version prior 2.2.
Upgrade django version to 2.2.
Adding py3.8 in tox and travis.
Bump the version.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [tox]
-envlist = {py27,py35}-django{111},py35-django{20,21,22}
+envlist = {py27,py35}-django{111},py35-django{20,21,22},py38-django{22}
```
Comment (UsamaSadiq) [inline tox.ini:2]: Shouldn't py27 be removed from here as well?

Comment (awais786) [reply tox.ini:2]: good catch.

File: .travis.yml [hunk 2]
```diff
@@ -3,8 +3,8 @@
 language: python
 
 python:
-  - 2.7
   - 3.5
+  - 3.8
 
 env:
   - TOXENV=django111
```
Comment (jmbowman) [inline .travis.yml:10]: Go ahead and remove the pre-2.2 test workers.

File: .travis.yml [hunk 3]
```diff
@@ -13,24 +13,31 @@ env:
...
     - python: 3.5
       env: TOXENV=quality
     - python: 3.5
       env: TOXENV=docs
+    - python: 3.8
+      env: TOXENV=quality
+    - python: 3.8
+      env: TOXENV=docs
+    - python: 3.8
+      env: TOXENV=django22
```
Comment (jmbowman) [inline .travis.yml:26]: You shouldn't need to add this one explicitly, should already be in the test matrix due to the top-level `python:` and `env:` lists.

File: requirements/constraints.txt [hunk 4]
```diff
@@ -8,7 +8,12 @@
...
-futures ; python_version == "2.7"
 
 
+# Stay on an LTS release
+django<2.3              # Web application framework
+
+
+
+# pip20+ requires for greater version of pip-tools. But installing 19.3.1 in all our containers and VMs.
+pip-tools==4.5.1
```
Comment (jmbowman) [inline requirements/constraints.txt:19]: I've applied a change to the `make upgrade` Jenkins jobs that should let us upgrade pip-tools in library packages like this one: https://github.com/edx/jenkins-job-dsl/pull/953 .  We probably want to keep it pinned in IDAs until we switch the configuration repo to build Docker images with newer pip versions, though.

File: setup.py [hunk 5]
```diff
@@ -91,15 +91,10 @@ def is_requirement(line):
...
-        'Framework :: Django :: 2.0',
-        'Framework :: Django :: 2.1',
         'Framework :: Django :: 2.2',
         'Intended Audience :: Developers',
         'License :: OSI Approved :: GNU Affero General Public License v3 or later (AGPLv3+)',
         'Natural Language :: English',
-        'Programming Language :: Python :: 2',
-        'Programming Language :: Python :: 2.7',
         'Programming Language :: Python :: 3',
         'Programming Language :: Python :: 3.5',
```
Comment (jmbowman) [inline setup.py:99]: Please add Python 3.8 here, and go ahead and make a new release so this gets updated on PyPI also.

File: tox.ini [hunk 6]
```diff
@@ -1,5 +1,5 @@
 [tox]
-envlist = {py27,py35}-django{111},py35-django{20,21,22}
+envlist = py35-django{111,20,21,22},py38-django{22}
```
Comment (jmbowman) [inline tox.ini:2]: Go ahead and remove the old Django versions.

File: .travis.yml [hunk 7]
```diff
@@ -3,34 +3,17 @@
...
-matrix:
-  exclude:
-    - python: 2.7
-      env: TOXENV=django20
-    - python: 2.7
-      env: TOXENV=django21
-    - python: 2.7
-      env: TOXENV=django22
-  include:
-    - python: 3.5
```
Comment (UsamaSadiq) [inline .travis.yml:24]: You should move the `quality` and `docs` in the `env` section above.

**Pull request merged**